base_layer/wallet/src/output_manager_service/storage/sqlite_db.rs

```diff
@@ -0,0 +1,884 @@
+        Ok(())
+    }
+}
+
+/// A utility function to construct a PendingTransactionOutputs structure for a TxId, set of Outputs and a Timestamp
+fn pending_transaction_outputs_from_sql_outputs(
+    tx_id: &TxId,
+    timestamp: &NaiveDateTime,
+    outputs: Vec<OutputSql>,
+) -> Result<PendingTransactionOutputs, OutputManagerStorageError>
+{
+    let mut outputs_to_be_spent = Vec::new();
+    let mut outputs_to_be_received = Vec::new();
+    for o in outputs {
+        if o.to_be_received == 1i32 {
+            outputs_to_be_received.push(UnblindedOutput::try_from(o.clone())?);
+        } else if o.to_be_received == 0i32 {
+            outputs_to_be_spent.push(UnblindedOutput::try_from(o.clone())?);
+        }
+    }
+
+    Ok(PendingTransactionOutputs {
+        tx_id: tx_id.clone() as u64,
+        outputs_to_be_spent,
+        outputs_to_be_received,
+        timestamp: timestamp.clone(),
+    })
+}
+
+/// This struct represents an Output in the Sql database. A distinct struct is required to define the Sql friendly
+/// equivalent datatypes for the members.
+#[derive(Clone, Debug, Queryable, Insertable, PartialEq)]
+#[table_name = "outputs"]
+struct OutputSql {
+    spending_key: Vec<u8>,
+    value: i64,
+    flags: i32,
+    maturity: i64,
+    spent: i32,
+    to_be_received: i32,
+    encumbered: i32,
+    tx_id: Option<i64>,
+}
+
+impl OutputSql {
+    pub fn new(
+        output: UnblindedOutput,
+        spent: bool,
+        to_be_received: bool,
+        encumbered: bool,
+        tx_id: Option<TxId>,
+    ) -> Self
+    {
+        Self {
+            spending_key: output.spending_key.to_vec(),
+            value: (u64::from(output.value)) as i64,
+            flags: output.features.flags.bits() as i32,
+            maturity: output.features.maturity as i64,
+            spent: spent as i32,
+            to_be_received: to_be_received as i32,
+            encumbered: encumbered as i32,
+            tx_id: tx_id.map(|i| i as i64),
+        }
+    }
+
+    /// Write this struct to the database
+    pub fn commit(
+        &self,
+        conn: &PooledConnection<ConnectionManager<SqliteConnection>>,
+    ) -> Result<(), OutputManagerStorageError>
+    {
+        diesel::insert_into(outputs::table).values(self.clone()).execute(conn)?;
+        Ok(())
+    }
+
+    /// Return all unencumbered outputs
+    #[cfg(test)]
+    pub fn index(
+        conn: &PooledConnection<ConnectionManager<SqliteConnection>>,
+    ) -> Result<Vec<OutputSql>, OutputManagerStorageError> {
+        Ok(outputs::table
+            .filter(outputs::encumbered.eq(false as i32))
+            .load::<OutputSql>(conn)?)
+    }
+
+    /// Return all unencumbered outputs with the specified spent status
+    pub fn index_spent(
+        spent: bool,
+        conn: &PooledConnection<ConnectionManager<SqliteConnection>>,
+    ) -> Result<Vec<OutputSql>, OutputManagerStorageError>
+    {
+        Ok(outputs::table
+            .filter(outputs::encumbered.eq(false as i32))
```

Comment:
Is there a smaller datatype that can be used to represent a bool, maybe a u8?

Reply:
Afraid not in Sqlite